doc/model.rst

```diff
@@ -0,0 +1,683 @@
+        from smartsim.ml.tf import serialize_model
+        n = Net()
+        input_shape = (3,3,1)
+        inputs = Input(input_shape)
+        outputs = n(inputs)
+        model = keras.Model(inputs=inputs, outputs=outputs, name=n.name)
+
+        return serialize_model(model)
+
+    # Get and save TF model
+    model, inputs, outputs = create_tf_cnn()
+
+**SmartSim Model Integration:**
+
+Assuming an initialized ``Model`` named `smartsim_model`, we specify the
+following parameters to the ``Model.add_ml_model()`` function:
+
+.. code-block:: python
+
+    smartsim_model.add_ml_model(name="cnn", backend="TF", model=model, device="GPU", devices_per_node=2, first_device=0, inputs=inputs, outputs=outputs)
+
+In this integration, we provide the following details:
+
+-  `name`: "cnn" - A key to uniquely identify the model within the database.
+-  `backend`: "TF" - Indicating that the model is a TensorFlow model.
+-  `model`: model - The in-memory representation of the TensorFlow model.
+-  `device`: "GPU" - Specifying the device for ML model execution.
+-  `devices_per_node`: 2 - Use two GPUs per node.
+-  `first_device`: 0 - Start with 0 index GPU.
+-  `inputs`: inputs - The names of the model inputs.
+-  `outputs`: outputs - The names of the model outputs.
+
+When the Model is started via ``Experiment.start()``, the ML model will be loaded to the
+standard orchestrator that is launched prior to the start of the Model.
+
+---------------------
+TorchScript functions
+---------------------
+Users can instruct SmartSim to upload TorchScript functions to the database
+at runtime. Script functions are loaded into
+standard orchestrators prior to the execution of Models. If using a
+colocated orchestrator, use the ``Model.add_script()`` function.
+Users have the flexibility to choose
+between `"GPU"` or `"CPU"` for device selection, and in environments with multiple
+devices, specific device numbers can be specified via `devices_per_node`.
+
+When specifying a TF function using ``Model.add_function()``, the
+following arguments are offered:
+
+- name (str): key to store function under
+- function (str, optional): TorchScript function code
+- device (str, optional): device for script execution, defaults to “CPU”
+- devices_per_node (int): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- first_device (int): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+
+Example: Loading an TensorFlow Function to the Model
+----------------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an TensorFlow function into the database at model runtime. It's
+important to note the function, ``Model.add_function()`` is supported
+for non-colocated deployments and during a colocated deployment, ``Model.add_script()``
+should be used.
```

Comment:
```
.. note::
- Use `Model.add_script` with a colocated deployment
- Use `Model.add_function` on standard model deployment
```

Reply:
I really love saying its important to note